File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let n = nums.length; 
    let max_ele = nums[0];
    for(let i=1;i<n;i++){
        if(nums[i-1] < nums[i]){
            max_ele = nums[i];
            i++;
        }
    }
    return max_ele;
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        max_ele = nums[0]
        count = 1
        for i in range(1, len(nums)):
            if(count == 0):
                count += 1
                max_ele = nums[i]
            elif max_ele == nums[i]:
                count += 1
            else:
                count -= 1 
        return max_ele
        